backend/sportsmatch/src/main/java/com/sportsmatch/auth/AuthConfig.java

```diff
@@ -19,6 +19,12 @@ public class AuthConfig {
 
   private final UserRepository userRepository;
 
+  /**
+   * Returns a UserDetailsService that fetches user details by email from a UserRepository.
+   *
+   * @return UserDetailsService instance.
+   * @throws UsernameNotFoundException If the user is not found.
```

Comment:
this method is a bit confusing but it doesn't throw an exception
it returns a UserDetailsService that has a method which accepts username and then possibly throws an exception

Reply:
originally it was like this
![image](https://github.com/MatejFrnka/sportsmatch/assets/134836290/6d9cdf8a-6b2e-4bda-925a-14bc45ac15ae)

but Intellij suggested the lambda that Im currently using and put the throw exception here I believe
![image](https://github.com/MatejFrnka/sportsmatch/assets/134836290/997c1363-ca1c-4c63-b4c7-d7dd9e371abb)